Dockerfile

```diff
@@ -0,0 +1,15 @@
+# syntax=docker/dockerfile:1
+FROM ubuntu:23.10
+WORKDIR /home/root
+RUN apt-get update && apt-get install -y --no-install-recommends \
+    python3 pip wget build-essential
+COPY requirements.txt ./
+RUN pip install -r requirements.txt --break-system-packages
+ARG MT_VERSION=4.6.0 # Version of memtester to build and run
+ENV MT_NAME memtester-${MT_VERSION}
+ENV MT_ARCHIVE ${MT_NAME}.tar.gz
+RUN wget https://pyropus.ca./software/memtester/old-versions/${MT_ARCHIVE} && \
+    tar -xf ${MT_ARCHIVE} && cd ${MT_NAME} && make && mv memtester .. && \
```

Comment:
This is already being compiled from source. Notice the `make` command at line 12. Or maybe I misunderstood the question...

Reply:
sorry, i wasnt very clear. What i was thinking here is whether this source is trusted enough to download it and compile as is.
The other question is if this only compiles x86_64 in this particular setup or can we also make/have some cross-chain for stuff like aarch64 (context: there's a team interested in running this against mobile devices)

Reply:
>whether this source is trusted enough to download it and compile as is

The source is the memtester author's web site. Can it be absolutely trusted? Well, even if the author does not have ill intent, we do not know what security policies they follow to maintain their server. On the other hand, the code is built and run in a Docker container, which provides an extra security layer. Whether Docker itself can be considered secure is a separate topic, but this is what the official guide says:

>Docker containers are, by default, quite secure; especially if you run your processes as non-privileged users inside the container.

>You can add an extra layer of safety by enabling AppArmor, SELinux, GRSEC, or another appropriate hardening system.

As for cross compilation, I fail to see how this can be helpful. The main purpose of this container is to test whether a particular memtester version causes parsing errors. Are we going to see any differences of memtester's output format across different platforms? Well, there is a chance, but I do not remember seeing any platform-dependent code, related to the output in memtester's src. However, if we absolutely need to verify platform compatibility, we can always run this container on the target platform. The [Docker Ubuntu image](https://hub.docker.com/_/ubuntu) seems to support aarch64, so we can find an ARM machine with Docker and try it.

WDYT?

Reply:
Moving `Dockerfile` to a subdirectory will move some files such as `requierements.txt` out of its context. Not sure how to solve this problem.